bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryDirectDataWriterHelper.java

```diff
@@ -253,6 +259,13 @@ public long finalizeStream() throws IOException {
     if (this.protoRows.getSerializedRowsCount() != 0) {
       sendAppendRowsRequest();
     }
+    try {
+      ApiFutures.allAsList(appendRowsFutures).get();
+    } catch (InterruptedException | ExecutionException e) {
+      throw new BigQueryConnectorException("Could not retrieve AppendRowsResponse", e);
```

Comment:
Should we log and throw? Also include some debug information along with the message. eg: tablePath

Reply:
This change has been made in https://github.com/GoogleCloudDataproc/spark-bigquery-connector/pull/1057